pkg/csi/service/wcpguest/controller.go

```diff
@@ -566,59 +566,83 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 
-	vmList := &vmoperatortypes.VirtualMachineList{}
-	err = c.vmOperatorClient.List(ctx, vmList, client.InNamespace(c.supervisorNamespace))
-	if err != nil {
-		msg := fmt.Sprintf("failed to list virtualmachines with error: %+v", err)
-		log.Error(msg)
-		return nil, status.Error(codes.Internal, msg)
-	}
+	if !c.coCommonInterface.IsFSSEnabled(ctx, common.OnlineVolumeExtend) {
+		vmList := &vmoperatortypes.VirtualMachineList{}
+		err = c.vmOperatorClient.List(ctx, vmList, client.InNamespace(c.supervisorNamespace))
+		if err != nil {
+			msg := fmt.Sprintf("failed to list virtualmachines with error: %+v", err)
+			log.Error(msg)
+			return nil, status.Error(codes.Internal, msg)
+		}
 
-	for _, vmInstance := range vmList.Items {
-		for _, vmVolume := range vmInstance.Status.Volumes {
-			if vmVolume.Name == volumeID && vmVolume.Attached {
-				msg := fmt.Sprintf("failed to expand volume: %q. Volume is attached to pod. Only offline volume expansion is supported", volumeID)
-				log.Error(msg)
-				return nil, status.Error(codes.FailedPrecondition, msg)
+		for _, vmInstance := range vmList.Items {
+			for _, vmVolume := range vmInstance.Status.Volumes {
+				if vmVolume.Name == volumeID && vmVolume.Attached {
+					msg := fmt.Sprintf("failed to expand volume: %q. Volume is attached to pod. Only offline volume expansion is supported", volumeID)
+					log.Error(msg)
+					return nil, status.Error(codes.FailedPrecondition, msg)
+				}
 			}
 		}
 	}
 
 	// Retrieve Supervisor PVC
-	pvc, err := c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Get(ctx, volumeID, metav1.GetOptions{})
+	svPVC, err := c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Get(ctx, volumeID, metav1.GetOptions{})
 	if err != nil {
 		msg := fmt.Sprintf("failed to retrieve supervisor PVC %q in %q namespace. Error: %+v", volumeID, c.supervisorNamespace, err)
 		log.Error(msg)
 		return nil, status.Error(codes.Internal, msg)
 	}
 
-	newQty := resource.NewQuantity(volSizeBytes, resource.Format(resource.BinarySI))
-	curQty := pvc.Spec.Resources.Requests[corev1.ResourceName(corev1.ResourceStorage)]
-	if (&curQty).Cmp(*newQty) == -1 {
-		// Update requested storage in PVC spec
-		pvcClone := pvc.DeepCopy()
-		pvcClone.Spec.Resources.Requests[corev1.ResourceName(corev1.ResourceStorage)] = *newQty
-		// Make a call to SV ControllerExpandVolume
-		log.Debugf("Increasing the size of supervisor PVC %s in namespace %s to %s", volumeID, c.supervisorNamespace, newQty.String())
-		pvc, err = c.supervisorClient.CoreV1().PersistentVolumeClaims(c.supervisorNamespace).Update(ctx, pvcClone, metav1.UpdateOptions{})
+	waitForSvPvcCondition := true
+	gcPvcRequestSize := resource.NewQuantity(volSizeBytes, resource.Format(resource.BinarySI))
+	svPvcRequestSize := svPVC.Spec.Resources.Requests[corev1.ResourceName(corev1.ResourceStorage)]
+	// Check if GC PVC request size is greater than SV PVC request size
+	if (gcPvcRequestSize).Cmp(svPvcRequestSize) == 1 {
```

Comment:
How about switching on `(gcPvcRequestSize).Cmp(svPvcRequestSize)`?

Reply:
Could you explain how is that better than this logic?

Reply:
Logic is the same, but its easier to read. This PR will flag if-else-else blocks - https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/387

Reply:
Thanks for that, changed it.